### ISSUE TITLE
LSP rename should collect non-LSP references before the rename and invoke `handleElementRename()` on them after successful rename

### ISSUE THREAD

**Issue by SCWells72** [redhat-developer/lsp4ij#670]
References can be added to LSP4IJ-managed files by non-LSP4IJ plugins, and those references need to know when a target has been renamed via LSP so they can be updated accordingly. I _think_ all that's needed is to collect references to the pseudo-element that's being renamed before the LSP rename, and then on success, iterate through those references and invoke `handleElementRename(newName)` on each so that things are kept consistent.

**Comment by angelozerr** [comment]
Have you a usecase which causes problem with rename?

**Comment by SCWells72** [comment]
> Have you a usecase which causes problem with rename?

Yes, absolutely. My own plugin adds references from HTML braced expressions to properties/methods of the associated JavaScript/TypeScript via a simple template binding syntax, e.g.,:

```html
    <record-edit-form
        record-id={recordId}
        onsubmit={handleSubmit}
        onerror={handleError}>
        ...
    </record-edit-form>
```

```typescript
export default class FormController {
    recordId: string;

    handleSubmit() {
        ...
    }


    handleError() {
        ...
    }
}
```

When, for example, `recordId` is renamed in the TypeScript file, all TypeScript/JavaScript references are properly updated, but the `{recordId}` reference in the HTML file is not resulting in an inconsistent state.

`PsiReference#handleElementRename()` exists explicitly to allow external references to respond to rename events accordingly. Those references just need to be included in the action since it's happening external to the IDE's own refactoring framework.

I'm happy to field this one if you'd like to assign it to me.

**Comment by angelozerr** [comment]
> I'm happy to field this one if you'd like to assign it to me.

Yes please. Thanks!

**Comment by SCWells72** [comment]
See PR #674.
